kafka-session-window-test/src/main/java/com/hazelcast/jet/tests/kafka/KafkaSessionWindowTest.java

```diff
@@ -105,18 +112,36 @@ public void test(HazelcastInstance client, String name) throws Exception {
         Job verificationJob = client.getJet().newJob(verificationPipeline(), verificationJobConfig);
 
         long begin = System.currentTimeMillis();
-        while (System.currentTimeMillis() - begin < durationInMillis) {
-            MINUTES.sleep(1);
-            assertFalse(producerFuture.isDone());
-            JobStatus status = getJobStatusWithRetry(verificationJob);
-            if (status != STARTING && status != RUNNING) {
-                throw new AssertionError("Job is failed, jobStatus: " + status);
+        String previousProcessedUuid = "initialValue";
+        IMap<String, String> itemProcessedMap = client.getMap(ITEM_PROCESSED_MAP);
+        itemProcessedMap.put(ITEM_PROCESSED_MAP_KEY, previousProcessedUuid);
+        int iteration = 0;
+        try {
+            while (System.currentTimeMillis() - begin < durationInMillis) {
+                MINUTES.sleep(1);
+                assertFalse(producerFuture.isDone());
+                JobStatus status = getJobStatusWithRetry(verificationJob);
+                if (status != STARTING && status != RUNNING) {
+                    throw new AssertionError("Job is failed, jobStatus: " + status);
+                }
+
+                // check whether verification pipeline actually processed some items
+                // start to check this after 5minutes
+                if (iteration >= 4) {
```

Comment:
```suggestion
                if (System.currentTimeMillis() - begin > Duration.ofMinutes(5).toMillis()) {
```
What if we wanted to change the time between iterations but keep this as 5 minutes? Something like the above would de-couple these and remove the need for the `iteration` variable (if you commit this remember to remove the `iteration` variable and `else` branch).